### ISSUE TITLE
Build failure on clang 3.8

### ISSUE THREAD

**Issue by mazimkhan** [ARMmbed/mbedtls#1072]
### Description
- Type: Enhancement
- Priority: Minor

Build fails with clang version 3.8 that is latest on Ubuntu 16.10 and Debian 9.

```
mazimkhan@mazimkhan-VirtualBox:~/github/ARMmbed/mbedtls-asan$ CC=/usr/bin/clang cmake -D CMAKE_BUILD_TYPE:String=Asan .
-- The C compiler identification is Clang 3.8.1
-- Check for working C compiler: /usr/bin/clang
-- Check for working C compiler: /usr/bin/clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found Perl: /usr/bin/perl (found version "5.22.2") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/mazimkhan/github/ARMmbed/mbedtls-asan
mazimkhan@mazimkhan-VirtualBox:~/github/ARMmbed/mbedtls-asan$ make
Scanning dependencies of target mbedcrypto
[  0%] Building C object library/CMakeFiles/mbedcrypto.dir/aes.c.o
clang: error: argument '-fno-sanitize-recover' is deprecated, use '-fno-sanitize-recover=undefined,integer' instead
library/CMakeFiles/mbedcrypto.dir/build.make:62: recipe for target 'library/CMakeFiles/mbedcrypto.dir/aes.c.o' failed
make[2]: *** [library/CMakeFiles/mbedcrypto.dir/aes.c.o] Error 1
CMakeFiles/Makefile2:327: recipe for target 'library/CMakeFiles/mbedcrypto.dir/all' failed
make[1]: *** [library/CMakeFiles/mbedcrypto.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2

```

---------------------------------------------------------------
## Bug

**OS**  
linux

**mbed TLS build:**  
Version: development HEAD
OS version: Ubuntu 16.10

**Expected behavior**   
Should build successfully with newer clang versions.

**Actual behavior**  

**Steps to reproduce**  
Install/Upgrade to newest clang version (comes out to be 3.8)
Run commands in clean mbedtls repo
```
CC=/usr/bin/clang cmake -D CMAKE_BUILD_TYPE:String=Asan .
make
```

----------------------------------------------------------------
## Enhancement\Feature Request

**Justification - why does the library need this feature?**  
Newer versions of clang are now available on Linux distros.

**Suggested enhancement**  

-----------------------------------------------------------------

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTSSL-1698

**Comment by sbutcher-arm** [comment]
Fix for `development` now merged with PR #1075, but `mbedtls-2.7` and `mbedtls-2.1` are outstanding.

**Comment by sbutcher-arm** [comment]
Now merged PR's #1075 for `development` and #1648 for `mbedtls-2.1`.

This issue can now be closed.
